src/components/IGVBrowser.js

```diff
@@ -1,80 +1,115 @@
 import _ from 'lodash/fp'
 import { Fragment, useRef, useState } from 'react'
 import { div, h } from 'react-hyperscript-helpers'
-import { requesterPaysWrapper } from 'src/components/bucket-utils'
-import { Link } from 'src/components/common'
+import { ButtonOutline, Link } from 'src/components/common'
 import { getUserProjectForWorkspace, parseGsUri } from 'src/components/data/data-utils'
 import { centeredSpinner, icon } from 'src/components/icons'
+import IGVAddTrackModal from 'src/components/IGVAddTrackModal'
+import RequesterPaysModal from 'src/components/RequesterPaysModal'
 import { Ajax, saToken } from 'src/libs/ajax'
 import colors from 'src/libs/colors'
-import { reportError } from 'src/libs/error'
-import { useCancellation, useOnMount, withDisplayName } from 'src/libs/react-utils'
+import { reportError, withErrorReporting } from 'src/libs/error'
+import { useCancellation, useOnMount } from 'src/libs/react-utils'
 import { knownBucketRequesterPaysStatuses, requesterPaysProjectStore } from 'src/libs/state'
 import * as Utils from 'src/libs/utils'
 
 
 // format for selectedFiles prop: [{ filePath, indexFilePath } }]
-const IGVBrowser = _.flow(
-  withDisplayName('IGVBrowser'),
-  requesterPaysWrapper({ onDismiss: ({ onDismiss }) => onDismiss() })
-)(({ selectedFiles, refGenome: { genome, reference }, workspace, onDismiss, onRequesterPaysError }) => {
+const IGVBrowser = ({ selectedFiles, refGenome: { genome, reference }, workspace, onDismiss }) => {
   const containerRef = useRef()
   const signal = useCancellation()
   const [loadingIgv, setLoadingIgv] = useState(true)
   const igvLibrary = useRef()
+  const igvBrowser = useRef()
+  const [requesterPaysModal, setRequesterPaysModal] = useState(null)
+  const [showAddTrackModal, setShowAddTrackModal] = useState(false)
```

Comment:
Our convention for clustering component-level `const`s has been to put `useState`s together on the top, which I find helpful because often I glance at state variables first when trying to understand a component.

Reply:
Sorted hooks in https://github.com/DataBiosphere/terra-ui/pull/3282/commits/1ca56d704aaf9adf87844dd9da5649eda2494e62. I think there's also an argument to be made for grouping based on concern vs type of hook.

Reply:
Sure. Perhaps a good topic to discuss as part of coding guidelines.